framework/web/Request.php

```diff
@@ -749,13 +749,18 @@ public function getHostInfo()
                 $this->_hostInfo = $http . '://' . trim(explode(',', $this->headers->get('X-Forwarded-Host'))[0]);
             } elseif ($this->headers->has('X-Original-Host')) {
                 $this->_hostInfo = $http . '://' . trim(explode(',', $this->headers->get('X-Original-Host'))[0]);
-            } elseif ($this->headers->has('Host')) {
-                $this->_hostInfo = $http . '://' . $this->headers->get('Host');
-            } elseif (isset($_SERVER['SERVER_NAME'])) {
-                $this->_hostInfo = $http . '://' . $_SERVER['SERVER_NAME'];
-                $port = $secure ? $this->getSecurePort() : $this->getPort();
-                if (($port !== 80 && !$secure) || ($port !== 443 && $secure)) {
-                    $this->_hostInfo .= ':' . $port;
+            } else {
+                if ($this->headers->has('Host')) {
+                    $this->_hostInfo = $http . '://' . $this->headers->get('Host');
+                } elseif (isset($_SERVER['SERVER_NAME'])) {
```

Comment:
use `filter_has_var(INPUT_SERVER, 'SERVER_NAME')` and `filter_input(INPUT_SERVER, 'SERVER_NAME')` instead direct access to globals: if you remove or change `$_SERVER['SERVER_NAME']`, `filter_*` still return initial value

Reply:
@bizley, What do you think?

Reply:
Yes, we could use `filter_input`.

Reply:
@lesha724 @bizley check tests, i don't remember how it work in phpunit less 7.4,